src/features/auth/signup/PhoneValidation/PhoneValidation.tsx

```diff
@@ -1,51 +1,242 @@
+        <ModalContent>
+          <Paragraphe>
+            <Typo.Body color={ColorsEnum.GREY_DARK}>
+              {t`Pour sécuriser l'accès à ton pass nous avons besoin de valider ton numéro.`}
+            </Typo.Body>
+          </Paragraphe>
+          <Spacer.Column numberOfSpaces={8} />
+          <PhoneNumberInput>
+            <CountryPickerPressable onPress={openCountryPickerModal}>
+              <CountryPicker
+                countryCode={countryCode}
+                countryCodes={ALLOWED_COUNTRY_CODES}
+                onSelect={onSelectCountryCode}
+                translation={'fra'}
+                theme={{
+                  ...DEFAULT_THEME,
+                  fontFamily: 'Montserrat-Bold',
+                  fontSize: getSpacing(3.75),
+                }}
+                withCallingCode
+                withCallingCodeButton
+                modalProps={{ visible: isCountryPickerVisible }}
+                onClose={closeCountryPickerModal}
+                onOpen={openCountryPickerModal}
+              />
+              <Animated.View
+                style={{ transform: [{ rotateZ: `${Math.PI / 2}rad` }] }}
+                testID="accordionArrow">
+                <ArrowNext size={getSpacing(6)} />
+              </Animated.View>
+            </CountryPickerPressable>
+            <Spacer.Row numberOfSpaces={4} />
+            <TextInput
+              autoCapitalize="none"
+              isError={false}
+              keyboardType="number-pad"
+              onChangeText={onChangeText}
+              placeholder={t`6 12 34 56 78`}
+              textContentType="telephoneNumber"
+            />
+          </PhoneNumberInput>
+          {invalidPhoneNumberMessage ? (
+            <React.Fragment>
+              <InputError visible messageId={invalidPhoneNumberMessage} numberOfSpacesTop={3} />
+              <Spacer.Column numberOfSpaces={5} />
+            </React.Fragment>
+          ) : (
+            <Spacer.Column numberOfSpaces={8} />
+          )}
+          <ButtonPrimary
+            title={getButtonTitle()}
+            disabled={!isContinueButtonEnabled}
+            onPress={requestSendPhoneValidationCode}
+          />
+        </ModalContent>
+      </BottomContentPage>
+      <QuitSignupModal
+        visible={quitSignupModalVisible}
+        resume={hideQuitSignupModal}
+        testIdSuffix="phone-number-quit-signup"
+        signupStep={SignupSteps.PhoneNumber}
       />
     </React.Fragment>
   )
 }
+
+const ModalContent = styled.View({
+  paddingTop: getSpacing(7),
+  alignItems: 'center',
+  width: '100%',
+  maxWidth: getSpacing(125),
+})
+
+const Paragraphe = styled.Text({
+  flexWrap: 'wrap',
+  flexShrink: 1,
+  textAlign: 'center',
+})
+
+const PhoneNumberInput = styled.View({
+  flexDirection: 'row',
+  paddingHorizontal: getSpacing(14),
+  alignItems: 'center',
+})
+
+const CountryPickerPressable = styled.TouchableOpacity({
+  flexDirection: 'row',
+})
+
+/**
+ * - contains 9 digits
```

Comment:
Oups j'avais du oublier de modifier le commentaire, tu peux mettre "contains minimum 6 and maximum 10 digits" ?

Reply:
OK